lib/rules/no-plusplus.js

```diff
@@ -46,12 +46,28 @@ module.exports = {
                 if (allowInForAfterthought && node.parent.type === "ForStatement") {
                     return;
                 }
+
+                let fixerFunc;
+
+                if (node.parent.type !== "ReturnStatement") {
+                    let operator = " -= 1";
+
+                    if (node.operator === "++") {
+                        operator = " += 1";
+                    }
+
+                    const fixedCode = node.argument.name + operator;
```

Comment:
I think using `node.argument.name` only works if `node.argument` is an `Identifier`; it will fail in other cases:

```js
foo.bar++;

// gets fixed to:

undefined += 1; // oops
```

I would recommend using `sourceCode.getText(node.argument)` instead of `node.argument.name`.